src/Simplify.cpp

```diff
@@ -3908,6 +3924,57 @@ class Simplify : public IRMutator {
             IRMutator::visit(op);
         }
     }
+    template <typename T>
+    Expr hoist_slice_vector(Expr a, Expr b) {
+        debug(4) << "Trying to hoist slice vector:\n... a = " << a << "\n... b = " << b << "\n";
+
+        const Call *call_a = a.as<Call>();
+        const Call *call_b = b.as<Call>();
+
+        internal_assert(call_a && call_b &&
+                        call_a->is_intrinsic(Call::slice_vector) &&
+                        call_b->is_intrinsic(Call::slice_vector));
+
+        if (!equal(call_a->args[1], call_b->args[1]) ||
+            !equal(call_a->args[2], call_b->args[2])) {
+            debug(4) << " ...unable to hoist - slice vector arguments don't match\n";
+            return T::make(a, b);
```

Comment:
The fall through in the visitor method does the same thing as well. May be we should fix that too?
https://github.com/halide/Halide/blob/lazy_slice_vectors/src/Simplify.cpp#L812